content/2016.12.how-to-develop-a-webapp-with-Kinto.md

```diff
@@ -0,0 +1,235 @@
+
+## API creation
+
+Hmm… what backend should we use? Django? Express? Firebase?… oh wait, why don’t use [Kinto](http://www.kinto-storage.org/)?
+
+Kinto is a perfect fit for this application as once installed, Kinto provides a flexible enough API to to create and handle other APIs.
+
+Let's see how to use Kinto for our instant chat WebApp.
+
+To show some pride, we will use an explicit subdomain for our main URL `https://kinto.discute.re/v1` to show our configuration rather than an agnostic https://api.discute.re/v1.
+
+Here are the different routes:
+
+- `/rooms/4815162342` becomes `/collections/4815162342` ;
+- `/rooms/4815162342/messages` becomes `/collections/4815162342/records` ;
+
+As we have seen before, we can add a new post to our backend with a simple POST request
+`/collections/4815162342/records`.
+
+A Kinto server can handle many Web applications through the concept of [*buckets*](https://kinto.readthedocs.io/en/latest/concepts.html#buckets-collections-and-records); a bucket can contain many *collections*, and a collection can contain many *records*, and records can contain arbitrary JSON data.
+
+![](https://kinto.readthedocs.io/en/latest/_images/concepts-general.png)
+{.image}
+
+For our application, we’ll create a `chat` bucket, which will contain as many collections as we want distinct chat rooms. Each room collection will itself represent chat messages through records.
+
+You can use [kinto-admin](https://github.com/Kinto/kinto-admin) plugin to easily manage your buckets, collections and records.
+
+![](https://d2mxuefqeaa7sj.cloudfront.net/s_4340C716162DF03B2B97575BFA438C542185694D96D86DB6EB44B989957A3ADE_1480537208814_file.png)
+{.image}
+
+So the request that will allow retrieving all messages from our example  `4815162342` room is:
+`GET https://kinto.discute.re/v1/buckets/chat/collections/4815162342/records`, and the request to post a new message is:
+`POST https://kinto.discute.re/buckets/chat/collections/4815162342/records`.
+
+It's very similar to our first envisioned API calls who were
+ `GET https://api.dicute.re/v1/rooms/4815162342/messages`
+ and `POST https://api.dicute.re/v1/rooms/4815162342/messages`.
+
+## Let’s install Kinto
+
+There are many ways to [install (or deploy) Kinto](https://kinto.readthedocs.io/en/stable/tutorials/install.html): [on Heroku](https://kinto.readthedocs.io/en/stable/tutorials/install.html#deploying-on-cloud-providers), [with Docker](https://kinto.readthedocs.io/en/stable/tutorials/install.html#using-docker), or [using a Python package](https://kinto.readthedocs.io/en/stable/tutorials/install.html#using-the-python-package).
+An ephemeral instance is also available on this server `[https://kinto.dev.mozaws.net/v1/](https://kinto.dev.mozaws.net/v1/.)`[.](https://kinto.dev.mozaws.net/v1/.)
```

Comment:
If you're using `this`, you should put a colon to separate the thing you're demonstrating from the rest of the sentence. Alternately, I think `on the server foo` is OK. I think the period at the end of the URL should be outside the parentheses? What's going on with the markdown here?